torch/_inductor/ir.py

```diff
@@ -6589,11 +6696,87 @@ class MutableBox(IRNode):
 
     data: IRNode
 
-    def __getattr__(self, name):  # type: ignore[no-untyped-def]
-        fn = getattr(self.data, name)
-        if callable(fn):
-            return fn
-        raise AttributeError(f"{type(self.data).__name__}.{name} not callable")
```

Comment:
@ezyang @aorenste I believe this function was the main reason typing ir.py was hard.  This little trick does save ~100 lines of boilerplate, but once I got rid of this (and a few `hasattr` -> `isinstance` changes in this PR) typing the rest of things seem straightforward.

Reply:
Yay!